cpp/text_module/algorithm/text.cpp

```diff
@@ -0,0 +1,34 @@
+#include "text.hpp"
+
+void Text::Join(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto list{arguments[0].ValueList()};
+    const auto delimiter{arguments[1].ValueString()};
+
+    std::string result;
+    if (list.Empty()) {
+      auto record = record_factory.NewRecord();
+      record.Insert(std::string(kResultJoin).c_str(), result);
+      return;
+    }
+
+    auto iterator = list.begin();
+    result += (*iterator).ValueString();
+
+    for (++iterator; iterator != list.end(); ++iterator) {
+      result += delimiter;
+      result += (*iterator).ValueString();
+    }
```

Comment:
Because you're already working with `std::string`, I recommend using `string::append` ([reference](https://cplusplus.com/reference/string/string/append/)).

Reply:
This actually isn't a part of this PR but resulted from merging with main, my bad.

I've actually checked and in most implementations += is defined as inline append call so should be the same.